internal/jobservice/eventstojobs/eventstojobs.go

```diff
@@ -0,0 +1,129 @@
+	return &EventsToJobService{
+		queue:                queue,
+		jobsetid:             jobsetid,
+		jobid:                jobid,
+		jobServiceConfig:     jobServiceConfig,
+		jobServiceRepository: jobServiceRepository}
+}
+
+func (eventToJobService *EventsToJobService) SubscribeToJobSetId(context context.Context) error {
+
+	return client.WithEventClient(&eventToJobService.jobServiceConfig.ApiConnection, func(c api.EventClient) error {
+		jobIdMap, err := eventToJobService.StreamCommon(c, context)
+		for key, element := range jobIdMap {
+			e := eventToJobService.jobServiceRepository.UpdateJobServiceDb(key, element)
+			if e != nil {
+				log.Error(e)
+				return e
+			}
+		}
+		return err
+	})
+}
+func (eventToJobService *EventsToJobService) GetStatusWithoutRedis(context context.Context, jobId string) (*js.JobServiceResponse, error) {
+	jobStatusForId := &js.JobServiceResponse{State: js.JobServiceResponse_JOB_ID_NOT_FOUND}
+	err := client.WithEventClient(&eventToJobService.jobServiceConfig.ApiConnection, func(c api.EventClient) error {
+		jobIdMap, err := eventToJobService.StreamCommon(c, context)
+		var ok bool
+		jobStatusForId, ok = jobIdMap[jobId]
+		if !ok {
+			jobStatusForId = &js.JobServiceResponse{State: js.JobServiceResponse_JOB_ID_NOT_FOUND}
+		}
+		return err
+	})
+	return jobStatusForId, err
+}
+func (eventToJobService *EventsToJobService) StreamCommon(c api.EventClient, ctx context.Context) (map[string]*js.JobServiceResponse, error) {
+	jobIdMap := make(map[string]*js.JobServiceResponse)
+	var fromMessageId string
+	// I found that GRPC will not allow you to run something in background and return a value back to caller.
+	// GRPC will cancel the context once your request returns.
+	// So we are going to introduce a timer for how long to subscribe to event.
+	// This will allow the rpc call to listen for all events in a given job-set
+	// But we will return after SubscribeJobSetTime s
+	ctxTimeout, cancel := context.WithTimeout(ctx, time.Duration(eventToJobService.jobServiceConfig.SubscribeJobSetTime)*time.Second)
+	defer cancel()
+	err := client.WithEventClient(&eventToJobService.jobServiceConfig.ApiConnection, func(c api.EventClient) error {
+		stream, err := c.GetJobSetEvents(ctx, &api.JobSetRequest{
+			Id:             eventToJobService.jobsetid,
+			Queue:          eventToJobService.queue,
+			Watch:          true,
+			FromMessageId:  fromMessageId,
+			ErrorIfMissing: false,
+		})
+		if err != nil {
+			log.Error(err)
+			return err
+		}
+		for {
+			msg, err := stream.Recv()
+			if err != nil {
+				log.Error(err)
+				return err
+			}
+			fromMessageId = msg.GetId()
+			currentJobId := api.JobIdFromApiEvent(msg.Message)
+			select {
+			case <-ctx.Done():
+				return ctx.Err()
+			// This allows us to subscribe for x amount of time
+			case <-ctxTimeout.Done():
+				log.Info("Hit a timeout")
+				return nil
+			default:
+			}
+			if !IsEventAJobResponse(*msg.Message) {
```

Comment:
wouldn't this be simpler as just 
```
if IsEventAJobResponse{
  ...
}
```
rather than declaring an empty if and then using the else for the logic